lightning/src/ln/channel.rs

```diff
@@ -1656,6 +1657,54 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		cmp::min(self.value_to_self_msat as i64 - self.get_outbound_pending_htlc_stats().1 as i64, 0) as u64)
 	}
 
+	fn commit_tx_fee_outbound_htlc(&self) -> u64 {
+		if !self.channel_outbound {
+			return 0;
+		} else {
+			let mut their_acked_htlcs = self.pending_inbound_htlcs.len() as u64;
+
+			for ref htlc in self.pending_outbound_htlcs.iter() {
+				match htlc.state {
+					OutboundHTLCState::Committed => their_acked_htlcs +=1,
+					OutboundHTLCState::RemoteRemoved {..} => their_acked_htlcs +=1,
+					OutboundHTLCState::LocalAnnounced {..} => their_acked_htlcs += 1,
+					_ => {},
+				}
+			}
+
+			for ref htlc in self.holding_cell_htlc_updates.iter() {
+				match htlc {
+					&&HTLCUpdateAwaitingACK::AddHTLC { .. } => their_acked_htlcs += 1,
+					_ => {},
+				}
+			}
+
+			return self.feerate_per_kw * (COMMITMENT_TX_BASE_WEIGHT + their_acked_htlcs * COMMITMENT_TX_WEIGHT_PER_HTLC);
+		}
+	}
+
+	fn commit_tx_fee_inbound_htlc(&self) -> u64 {
+		if self.channel_outbound {
+			return 0;
+		} else {
+			let mut their_acked_htlcs = self.pending_inbound_htlcs.len() as u64;
+
+			// When calculating the set of HTLCs which will be included in their next
+			// commitment_signed, all inbound HTLCs are included (as all states imply it will be
+			// included) and only committed outbound HTLCs, see below.
+			for ref htlc in self.pending_outbound_htlcs.iter() {
+				if let OutboundHTLCState::Committed = htlc.state {
```

Comment:
6ca654a

This maybe an issue if another implementation computes their balance differently and increase it by any announced-but-non-committed update_fulfill_htlc,  they may expect a new update_add_htlc to be accepted ?

And even if they have the same fee computation handling, the order in which we receive update_add_htlc, update_fulfill may change acceptance, the spec is doing assumption we rely on an ordered transport (TCP) but that's clearly something RL is not doing ?

I lean to be liberal on the receiver-side, and not close the channel for a too-expensive HTLC.

Reply:
I'm in favor of being liberal as well. We could also not close the channel but still refuse to add the HTLC?

To be clear, are you advocating for not counting any pending_outbound_htlcs against their commitment tx cost?

Reply:
No I would advocate for integrating any pending_outbound_htlcs::Committed and _also_ pending_outbound::RemoteRemoved to be sure HTLC acceptance is predictable independently of reception order, predictable at least from our viewpoint. 

Not counting them mean we may sign them a lower-feerate commitment transaction, which would hinder our security if we need their local commitment transaction to confirm.

So conservative on how we accept HTLC but liberal on how we deal with acceptance rejection.